src/test/java/pl/byczazagroda/trackexpensesappbackend/integration/FindAllWalletsByNameCaseSensitive.java

```diff
@@ -0,0 +1,81 @@
+import pl.byczazagroda.trackexpensesappbackend.model.Wallet;
+import pl.byczazagroda.trackexpensesappbackend.repository.FinancialTransactionRepository;
+import pl.byczazagroda.trackexpensesappbackend.repository.WalletRepository;
+
+import static org.hamcrest.Matchers.hasSize;
+import static org.springframework.test.web.servlet.result.MockMvcResultHandlers.print;
+
+public class FindAllWalletsByNameCaseSensitive extends BaseIntegrationTestIT {
+
+    static private final String WALLET_NAME = "wallet";
+    static private final String WALLET_NAME_TOO_LONG = "walletttttttttttttttttttttttttttttttttttttttttttttt";
+    @Autowired
+    WalletRepository walletRepository;
+
+    @Autowired
+    FinancialTransactionRepository financialTransactionRepository;
+
+    @BeforeEach
+    void clearDatabase() {
+        walletRepository.deleteAll();
+        financialTransactionRepository.deleteAll();
+    }
+
+    @DisplayName("Find all Wallets with corresponding search name, ignoring case")
+    @Test
+    void testFindAllWalletsByNameIgnoringCaseAPI_whenSearchNameIsProvided_thenShouldReturnAllWalletsWithSearchNameIgnoringCase()
+            throws Exception {
+        Wallet wallet1 = walletRepository.save(new Wallet("Wallet"));
+        Wallet wallet2 = walletRepository.save(new Wallet("wallet"));
+        Wallet wallet3 = walletRepository.save(new Wallet("WALLET"));
+        Wallet wallet4 = walletRepository.save(new Wallet("Bag"));
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/wallets/wallets/{name}", WALLET_NAME).accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
+                .andExpect(MockMvcResultMatchers.status().isOk())
+                .andExpect(MockMvcResultMatchers.jsonPath("$.*", hasSize(3)))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[0].id").value(wallet1.getId()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[1].id").value(wallet2.getId()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[2].id").value(wallet3.getId()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[0].name").value(wallet1.getName()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[1].name").value(wallet2.getName()))
+                .andExpect(MockMvcResultMatchers.jsonPath("$.[2].name").value(wallet3.getName()));
+
+        //check if database has 4 records
+        Assertions.assertEquals(4, walletRepository.count());
+    }
+
+    @DisplayName("When search name is too long then empty array and error - bad request should be returned")
+    @Test
+    void testFindAllWalletsByNameIgnoringCaseAPI_whenSearchNameTooLong_thenShouldReturnTEA003Error() throws Exception {
+        Wallet wallet1 = walletRepository.save(new Wallet("Wallet"));
+        mockMvc.perform(MockMvcRequestBuilders.get("/api/wallets/wallets/{name}", WALLET_NAME_TOO_LONG).accept(MediaType.APPLICATION_JSON))
+                .andDo(print())
```

Comment:
We need to consider it, but in my opinion this line would be great for debugging only

Reply:
Indeed, that was for debugging, to see visually what should be returned. It is not critical for the test, so if you want me to remove it I can do it